packages/logging/test/LoggerService.test.ts

```diff
@@ -0,0 +1,156 @@
+import createHttpError from "http-errors";
+import pino, { Logger } from "pino";
+import pinoFluentd from "pino-fluentd";
+import pretty from "pino-pretty";
+
+import { config } from "../src/config";
+import { LoggerService } from "../src/index";
+
+jest.mock("pino");
+jest.mock("pino-fluentd");
+jest.mock("pino-pretty");
+
+describe("LoggerService", () => {
+  let loggerService: LoggerService;
+  let mockLogger: jest.Mocked<Logger>;
+  const defaultLogFormat = config.STD_OUT_LOG_FORMAT;
+  const defaultFluentdTag = config.FLUENTD_TAG;
+  const defaultFluentdHost = config.FLUENTD_HOST;
+  const defaultFluentdPort = config.FLUENTD_PORT;
+
+
+  beforeEach(() => {
+    mockLogger = {
+      info: jest.fn(),
+      error: jest.fn(),
+      warn: jest.fn(),
+      debug: jest.fn(),
+      child: jest.fn().mockReturnThis()
+    } as unknown as jest.Mocked<Logger>;
+
+    (pino as unknown as jest.Mock).mockReturnValue(mockLogger);
+    loggerService = new LoggerService();
+  });
+
+  afterEach(() => {
+    config.STD_OUT_LOG_FORMAT = defaultLogFormat;
+    config.FLUENTD_TAG = defaultFluentdTag;
+    config.FLUENTD_HOST = defaultFluentdHost;
+    config.FLUENTD_PORT = defaultFluentdPort;
+    jest.clearAllMocks();
+  });
+
+  describe("initPino", () => {
+    it("should initialize pino with pretty formatting when STD_OUT_LOG_FORMAT is 'pretty'", () => {
+      config.STD_OUT_LOG_FORMAT = "pretty";
+      new LoggerService();
+      expect(pretty).toHaveBeenCalledWith({ sync: true });
+    });
+
+    it("should initialize pino without pretty formatting for other formats", () => {
+      config.STD_OUT_LOG_FORMAT = "json";
+      new LoggerService();
+      expect(pretty).not.toHaveBeenCalled();
+      expect(pino).toHaveBeenCalled();
+    });
+
+    it("should initialize fluentd if configuration is enabled", () => {
+      config.FLUENTD_HOST = "localhost";
+      config.FLUENTD_PORT = 24224;
+      config.FLUENTD_TAG = "app";
+
+      new LoggerService();
+      expect(pinoFluentd).toHaveBeenCalledWith({
+        tag: config.FLUENTD_TAG,
+        host: config.FLUENTD_HOST,
+        port: config.FLUENTD_PORT,
+        "trace-level": config.LOG_LEVEL
+      });
+    });
+
+    it("should not initialize fluentd if configuration is missing", () => {
+      config.FLUENTD_HOST = "";
+      config.FLUENTD_PORT = 0;
+      config.FLUENTD_TAG = "";
+
+      new LoggerService();
+      expect(pinoFluentd).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("Logging methods", () => {
```

Comment:
I suggest to adjust like so:

```typescript
const methods: (keyof Logger)[] = ["info", "error", "warn", "debug"];
  describe.each(methods)("prototype.%s", method => {
    const logMessage = "Test message";

    it(`should call pino.${method} on info method`, () => {
      loggerService[method](logMessage);
      expect(mockLogger[method]).toHaveBeenCalledWith(logMessage);
    });
  });
```

for this Logger would need to implement common interface, which can be done like so:

```typescript
import pino, { Logger as PinoLogger } from "pino";
export type Logger = Pick<PinoLogger, "info" | "error" | "warn" | "debug">;

export class LoggerService implements Logger {
```